### PR TITLE
EP-883: adds cli command for full re-sync of posts data

### PR DESCRIPTION
**Pull request by sergunich**
```sh
$ wp --debug=empire empire_sync_content --full --sleep-between=1 --batch-size=100
```

Bouns: handle cases with empty content in prefill/amp injection code

### CODE REVIEW

File: Empire/ContentSyncCommand.php [hunk 1]
```diff
@@ -34,30 +34,59 @@ public function run() {
...
+        if ( count( $post_ids ) ) {
+            foreach ( $post_ids as $post_id ) {
                 $post = \WP_Post::get_instance( $post_id );
                 if ( ! $post ) {
                     $this->empire->log( 'Post ' . $post_id . ' not found. Skipping...' );
                     continue;
                 }
                 $this->empire->syncPost( $post );
             }
+            return;
```
[the remark this answers is not in the document]
Comment (sergunich) [reply Empire/ContentSyncCommand.php:86]: Yeah, if there was passed a list of post IDs then we want to sync only those and do not get to the `syncContent` part

File: Empire/ContentSyncCommand.php [hunk 2]
```diff
@@ -34,30 +34,59 @@ public function run() {
...
+                (int)($opts['batch-size'] ?? 100),
+                (int)($opts['start-from'] ?? 0),
+                (int)($opts['sleep-between'] ?? 1),
+            );
+            $this->empire->log( 'Empire Sync: total_posts=' . $updated );
+            return;
+        }
+
+        $post_ids = array_filter(explode(',', ($opts['posts'] ?? '')));
+        if ( count( $post_ids ) ) {
```
[the remark this answers is not in the document]
Comment (sergunich) [reply Empire/ContentSyncCommand.php:77]: Yes, but if the list of post IDs was passed, then we want to return early, and that's why I'm using `if` here

[the remark this answers is not in the document]
Comment (rrdial) [reply Empire/ContentSyncCommand.php:77]: Ok, this makes sense.

File: Empire/Empire.php [hunk 3]
```diff
@@ -642,10 +677,22 @@ function_exists( 'App\get_article_author' )
                 $categories,
                 $tags
             );
+            $this->debug(
+                'Empire Sync: SUCCESS | ' . json_encode([
+                    'external_id' => $external_id,
+                    'url' => $canonical,
+                ])
+            );
```
[the remark this answers is not in the document]
Comment (sergunich) [reply Empire/Empire.php]: Handled this here https://github.com/empireio/wordpress-plugin/pull/26/

**Comment by sergunich** [comment]
Closing this PR in favor of these more granular PRs:
- EP-883: full re-sync of posts data (#27)
- Make logging more PSR-3 complaint (#26)
- EP-883: use 'filter' to get authors data for sync (#25)
- EP-883: allow to pass API URL with env var (#24)

**Pull request closed without merging**